### ISSUE TITLE
newlines added when editing a post with Firefox

### ISSUE THREAD

**Issue by kikelkik** [mybb/mybb#3768]
This bug seems to occur since the newest MyBB 1.8.21 release.
I've updated sceditor correctly and this bug is even reproducable on mybb.com community forum.
It might has got something to do with browser compatibilty, because in Chrome there is no problem but in Firefox (68.0.1 newest version) the error is there.

Steps to reproduce:
(1) Have a thread with some line breaks like:
1
2
3
4
5

(2) Save the post
(3) Edit the post with the editor (not quick edit)
(4) Set the cursor at the very begin of a line. For example right before the "3".
(5) Press the Backspace Key of your keyboard one time. Nothing despite of a mysterious hidden character seems to be deleted.
(6) Save the post.
(7) The post now looks like this:
1

2
3
4

5 

Despite of the edited line and one up and down, between every line there is generated an empty line. No matter how long the post is.
This often results in annoying post breaks when users are deleting some text/lines and these "magical invisible characters" stay there or are missing.

I hope you can reproduce the problem. Four other persons tested it for me and experienced the same bug with their Firefox browsers but not with Chrome.

**Comment by yuliu** [comment]
I have encountered this problem for several times, using Firefox on Windows. Looks like it relates to the SCEditor, that's why Quick Edit is doing well. Always use source editor should temporarily prevent it occurring, I suppose.

I'm waiting for 1.9's new editor, so I neglect this problem :rofl:

**Comment by rajat315315** [comment]
It seems I cannot reproduce this issue in Firefox 78.0.1
Is someone still able to reproduce?

**Comment by RaymanXY** [comment]
Still able to reproduce with Firefox 78.0.2 and MyBB 1.8.22. I will update soon to 1.8.23 and try again.

**Comment by kikelkik** [comment]
Still occurs in 1.8.24

**Comment by euantorano** [comment]
> Still occurs in 1.8.24

Yes, we've [forked SCEditor](https://github.com/mybb/sceditor) to try and fix issues like this in a future release. We have some other maintenance tasks to do on SCEditor too.

**Comment by TRSx80** [comment]
I think I might be experiencing this as well.

I had made a post on Pine64 forum (they use mybb), and it was [there](https://forum.pine64.org/showthread.php?tid=13220&pid=90446#pid90446) that their sysadmin (fire219) actually pointed me towards this issue.

As reported in that thread, I am using Firefox 78.7.0esr (64-bit) on Debian Testing.  The detailed description of my symptoms can be seen in my OP over there (along with a link to an actual post with an example), so I won't repeat them here.

Should I create an issue in the new forked repo?  Or right here is good enough?

**Comment by euantorano** [comment]
> I think I might be experiencing this as well.
> 
> I had made a post on Pine64 forum (they use mybb), and it was [there](https://forum.pine64.org/showthread.php?tid=13220&pid=90446#pid90446) that their sysadmin (fire219) actually pointed me towards this issue.
> 
> As reported in that thread, I am using Firefox 78.7.0esr (64-bit) on Debian Testing. The detailed description of my symptoms can be seen in my OP over there (along with a link to an actual post with an example), so I won't repeat them here.
> 
> Should I create an issue in the new forked repo? Or right here is good enough?

It might be a good idea to migrate this issue to the forked repository. I need to try and find some time to test this and see if I can find the cause.

**Comment by euantorano** [comment]
It might also be a good idea to see if this issue is fixed with some of the recent commits to [the main SCEditor repository](https://github.com/samclarke/SCEditor/commits/master) now that it seems to be receiving active development again...

I'll update our fork this weekend and try updating it in the core so we can see if it helps and if so it'll make its way into 1.8.26.

**Comment by Zathos91** [comment]
Any updates?

**Comment by endrift** [comment]
I'm hitting this on 1.8.33. It eventually slows down to a crawl after typing less than a whole sentence and makes posting impossible on Firefox. Which is a problem because I run this forum. Is there something I can test?

**Comment by arcalys** [comment]
Haven't been able to reproduce on several recent versions of Firefox and the two latest versions of SCEditor.
